tests/rptest/services/admin.py

```diff
@@ -267,7 +267,11 @@ def __init__(self, topic: str, namespace: str = "kafka"):
         self.topic = topic
 
     def as_dict(self):
-        return {'ns': self.ns, 'topic': self.topic}
+        ret = {'topic': self.topic}
+        # randomly omit default namespace
+        if self.ns != "kafka" or random.choice([True, False]):
+            ret['ns'] = self.ns
+        return ret
```

Comment:
this is very confusing for the end user of the `Admin` API, this is a public interface used by the other teams to implement tests, we shouldn't introduced randomness here, maybe we can do it as a wrapper in the test where it is used ?